cli/exchange/org.go

```diff
@@ -0,0 +1,165 @@
+package exchange
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/open-horizon/anax/cli/cliutils"
+	"github.com/open-horizon/anax/i18n"
+	"net/http"
+)
+
+type ExchangeOrgs struct {
+	Orgs      map[string]interface{} `json:"orgs"`
+	ListIndex int                    `json:"lastIndex"`
+}
+
+type OrgHeartbeat struct {
+	Adjust int `json:"intervalAdjustment"`
+	Max    int `json:"maxInterval"`
+	Min    int `json:"minInterval"`
+}
+
+type OrgExchangeReq struct {
+	Label       string        `json:"label,omitempty"`
+	Description string        `json:"description,omitempty"`
+	Heartbeat   *OrgHeartbeat `json:"heartbeatIntervals,omitempty"`
+}
+
+func OrgList(org, userPwCreds, theOrg string, long bool) {
+	// get message printer
+	msgPrinter := i18n.GetMessagePrinter()
+	exchUrlBase := cliutils.GetExchangeUrl()
+
+	// if org name is empty, list name of all orgs
+	var orgurl string
+	nameOnly := false
+	if theOrg == "" {
+		orgurl = "orgs"
+		nameOnly = true
+	} else {
+		orgurl = "orgs/" + theOrg
+	}
+
+	// get orgs
+	var orgs ExchangeOrgs
+	httpCode := cliutils.ExchangeGet("Exchange", exchUrlBase, orgurl, cliutils.OrgAndCreds(org, userPwCreds), []int{200, 404}, &orgs)
+	if httpCode == 404 {
+		cliutils.Fatal(cliutils.NOT_FOUND, msgPrinter.Sprintf("org '%s' not found.", theOrg))
+	}
+
+	// Print only the names if listing all orgs AND -l was not inputted
+	// in the case of -l being true, print all details regardless
+	if !long && nameOnly {
+		organizations := []string{}
+		for o := range orgs.Orgs {
+			organizations = append(organizations, o)
+		}
+
+		jsonBytes, err := json.MarshalIndent(organizations, "", cliutils.JSON_INDENT)
+		if err != nil {
+			cliutils.Fatal(cliutils.JSON_PARSING_ERROR, msgPrinter.Sprintf("failed to marshal 'exchange org list' output: %v", err))
+		}
+		fmt.Printf("%s\n", jsonBytes)
+	} else {
+		output := cliutils.MarshalIndent(orgs.Orgs, "exchange orgs list")
+		fmt.Println(output)
+	}
+}
+
+func OrgCreate(org, userPwCreds, theOrg string, label string, desc string, min int, max int, adjust int) {
+	// get message printer
+	msgPrinter := i18n.GetMessagePrinter()
+
+	// get credentials
+	cliutils.SetWhetherUsingApiKey(userPwCreds)
+
+	// check constraints on min, max, amnd adjust
+	if min < 0 || max < 0 || adjust < 0 {
```

Comment:
The values cannot be zero either. Please update this same check in the update path.

Reply:
Done. Changed the conditions.